xword.py

```diff
@@ -1,21 +1,45 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
-"""Crossword Solver Program"""
+"""
+Crossword Solver Program
 
-__author__ = "???"
+Description: displays possible words to fit crossword puzzle location based on
+dictionary of words
 
-# YOUR HELPER FUNCTION GOES HERE
+Author: Aaron Jackson
+Github: TimeApollo
+"""
+
+__author__ = "TimeApollo"
+import re
+
+
+def word_checker(in_word, words):
+    """checks in_word against words list, to see if any match pattern"""
+    """returns list as well as printing all matches"""
+    regex = in_word.replace(' ', '\\w')
+    word_list = []
+    for word in words:
+        if re.match(r'{0}'.format(regex), word) and len(in_word) is len(word):
```

Comment:
Don't use the `is` keyword to test if two variables have the same value, use `==` instead.  The `is` keyword is for testing whether two _objects_ are the same object.  See https://stackoverflow.com/questions/22885931/when-if-ever-to-use-the-is-keyword-in-python

Reply:
Changed to reflect comment. Thanks for showing me that. was getting confused when to use the 2.